servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DefaultDnsClientTest.java

```diff
@@ -123,10 +121,112 @@ public void unknownHostDiscover() throws Exception {
             assertThat(subscriber.activeCount(), equalTo(0));
             assertThat(subscriber.inactiveCount(), equalTo(0));
         } finally {
-            discoverer.closeAsync().toFuture().get();
+            filteredClient.closeAsync().toFuture().get();
         }
     }
 
+    @Test
+    public void singleSrvSingleADiscover() throws InterruptedException {
+        final String domain = "mysvc.apple.com";
+        final String targetDomain = "target.mysvc.apple.com";
+        final int targetPort = 9876;
+        recordStore.addSrvResponse(domain, targetDomain, 10, 10, targetPort);
+        recordStore.addResponse(targetDomain, A, nextIp());
+        final int expectedActiveCount = 1;
+        final int expectedInactiveCount = 0;
+
+        CountDownLatch latch = new CountDownLatch(expectedActiveCount + expectedInactiveCount);
+        AtomicReference<Throwable> throwableRef = new AtomicReference<>();
+        Publisher<ServiceDiscovererEvent<InetSocketAddress>> publisher = client.dnsSrvQuery(domain);
+        ServiceDiscovererTestSubscriber<InetSocketAddress> subscriber =
+                new ServiceDiscovererTestSubscriber<>(latch, throwableRef, expectedActiveCount);
+        toSource(publisher).subscribe(subscriber);
+
+        latch.await();
+        assertNull(throwableRef.get());
+        assertThat(subscriber.activeCount(), equalTo(expectedActiveCount));
+        assertThat(subscriber.inactiveCount(), equalTo(expectedInactiveCount));
+    }
+
+    @Ignore("Publisher#flatMapMerge required https://github.com/apple/servicetalk/pull/1011")
+    @Test
+    public void singleSrvMultipleADiscover() throws InterruptedException {
+        final String domain = "mysvc.apple.com";
+        final String targetDomain = "target.mysvc.apple.com";
+        final int targetPort = 9876;
+        recordStore.addSrvResponse(domain, targetDomain, 10, 10, targetPort);
+        recordStore.addResponse(targetDomain, A, nextIp(), nextIp());
+        final int expectedActiveCount = 2;
+        final int expectedInactiveCount = 0;
+
+        CountDownLatch latch = new CountDownLatch(expectedActiveCount + expectedInactiveCount);
+        AtomicReference<Throwable> throwableRef = new AtomicReference<>();
+        Publisher<ServiceDiscovererEvent<InetSocketAddress>> publisher = client.dnsSrvQuery(domain);
+        ServiceDiscovererTestSubscriber<InetSocketAddress> subscriber =
+                new ServiceDiscovererTestSubscriber<>(latch, throwableRef, expectedActiveCount);
+        toSource(publisher).subscribe(subscriber);
+
+        latch.await();
+        assertNull(throwableRef.get());
+        assertThat(subscriber.activeCount(), equalTo(expectedActiveCount));
+        assertThat(subscriber.inactiveCount(), equalTo(expectedInactiveCount));
+    }
+
+    @Test
+    public void multipleSrvSingleADiscover() throws InterruptedException {
```

Comment:
This can be done as a followup if you have verified that changes to SRV records are reflected in the lookup, but it will be good to have a unit test for it.

Reply:
good call, I added a test. The test class was introduced before we had testing utilities and is difficult to control/analyze. Here is an issue to analyze https://github.com/apple/servicetalk/issues/1022.